editor/animation_track_editor.cpp

```diff
@@ -5011,7 +5011,7 @@ void AnimationTrackEditor::_add_track(int p_type) {
 		return;
 	}
 	adding_track_type = p_type;
-	pick_track->popup_scenetree_dialog();
+	pick_track->popup_scenetree_dialog(nullptr, root, true);
```

Comment:
Something is wrong with the `root` variable. When you change the `root_node` property, you need to reselect the AnimationPlayer to update it.

Reply:
Yep, seems to be updated too rarely. This also causes the warning above to not always show.

Other places in the AnimationTrackEditor appear to query the AnimationPlayer via a singleton, so I'll just follow that logic to get the up-to-date root node.